ladybug/_datacollectionbase.py

```diff
@@ -0,0 +1,517 @@
+                    if raise_exception is True:
+                        error_msg = '{} Data Collection is not aligned with '\
+                            '{} Data Collection.'.format(
+                                first_coll.header.data_type, coll.header.data_type)
+                        raise ValueError(error_msg)
+                    return False
+        return True
+
+    @staticmethod
+    def _check_conditional_statement(statement, num_collections):
+        """Method to check conditional statements to be sure that they are valid.
+
+        Args:
+            statement: A conditional statement as a string (e.g. a>25 and a%5==0).
+                The variable should always be named as 'a' (without quotations).
+            num_collections: An integer representing the number of data collections
+                that the statement will be evaluating.
+
+        Return:
+            correct_var: A list of the correct variable names that should be
+                used within the statement (eg. ['a', 'b', 'c'])
+        """
+        # Determine what the list of variables should be based on the num_collections
+        correct_var = list(ascii_lowercase)[:num_collections]
+
+        # Clean out the operators of the statement
+        st_statement = BaseCollection._remove_operators(statement)
+        parsed_st = [s for s in st_statement if s.isalpha()]
+
+        # Perform the check
+        for var in parsed_st:
+            if var not in correct_var:
+                raise ValueError(
+                    'Invalid conditional statement: {}\n '
+                    'Statement should be a valid Python statement'
+                    ' and the variables should be named as follows: {}'.format(
+                        statement, ', '.join(correct_var))
+                    )
+        return correct_var
+
+    @staticmethod
+    def _remove_operators(statement):
+        """Remove logical operators from a statement."""
+        return statement.lower().replace("and", "").replace("or", "") \
+            .replace("not", "").replace("in", "").replace("is", "")
+
+    @staticmethod
+    def _replace_operators(statement):
+        """Replace logical operators of a statement with non-alphanumeric characters."""
+        return statement.lower().replace("and", "&&").replace("or", "||") \
+            .replace("not", "~").replace("in", "<<").replace("is", "$")
+
+    @staticmethod
+    def _restore_operators(statement):
+        """Restore python logical operators from previusly replaced ones."""
+        return statement.replace("&&", "and").replace("||", "or") \
+            .replace("~", "not").replace("<<", "in").replace("$", "is")
+
+    def _filter_by_statement(self, statement):
+        """Filter the data collection based on a conditional statement."""
+        self.__class__._check_conditional_statement(statement, 1)
+        statement = statement.replace('a', 'd')
+        _filt_values, _filt_datetimes = [], []
```

Comment:
You are right that the replacement process is unnecessary and just leftover from the previous code.  I also realize that it introduces bugs if someone passes a statement like "if a > 20 and a < 30" since the "a" in the "and" will get replaced.  I have taken this whole replacement out and that also fixes the bug.

I don't think that it is better to have the iteration over the values happen inside of the eval() and I also don't think that this is possible (it would have to be an exec() statement instead of eval()).  Multiple eval() calls definitely seems like the cleanest/safest way to do it (particularly now that I have made it such that eval can only act on the "a" variable and nothing else).  I should also say that iterating over eval is how you had originally written this method and I think this is the right way.